altos-rust/port/cortex-m0/src/peripheral/usart/control.rs

```diff
@@ -23,100 +55,204 @@ impl UsartControl {
-        self.cr1.set_word_length(length);
+    /// Makes call to control register one to set the Usart mode.
+    /// Possible Mode settings are: None, Receive, Transmit, All
+    pub fn set_mode(&mut self, mode: Mode) {
+        self.cr1.set_mode(mode);
     }
 
-    pub fn set_mode(&self, mode: Mode) {
-        self.cr1.set_mode(mode);
+    /// Makes call to control register one to enable the RXNE interrupt.
+    pub fn enable_receiver_not_empty_interrupt(&mut self) {
+        self.cr1.set_receiver_not_empty_interrupt(true);
+    }
+
+    /// Makes call to control register one to disable the RXNE interrupt.
+    pub fn disable_receiver_not_empty_interrupt(&mut self) {
+        self.cr1.set_receiver_not_empty_interrupt(false);
+    }
+
+    /// Makes call to control register one to enable the TC interrupt.
+    pub fn enable_transmit_complete_interrupt(&mut self) {
+        self.cr1.set_transmit_complete_interrupt(true);
+    }
+
+    /// Makes call to control register one to disable the TC interrupt.
+    pub fn disable_transmit_complete_interrupt(&mut self) {
+        self.cr1.set_transmit_complete_interrupt(false);
+    }
+
+    /// Makes call to control register one to enable the TXEIE interrupt.
+    pub fn enable_transmit_interrupt(&mut self) {
+        self.cr1.set_transmit_interrupt(true);
+    }
+
+    /// Makes call to control register one to disable the TXEIE interrupt.
+    pub fn disable_transmit_interrupt(&mut self) {
+        self.cr1.set_transmit_interrupt(false);
     }
 
-    pub fn set_parity(&self, parity: Parity) {
+    /// Makes call to control register one to set the parity.
+    /// Possible Parity settings are: None, Even, Odd
+    pub fn set_parity(&mut self, parity: Parity) {
         self.cr1.set_parity(parity);
     }
 
-    pub fn set_stop_bits(&self, length: StopLength) {
-        self.cr2.set_stop_bits(length);
+    /// Makes call to control register one to set the word length.
+    /// Possible WordLength settings are: Seven, Eight, Nine
+    pub fn set_word_length(&mut self, length: WordLength) {
+        self.cr1.set_word_length(length);
     }
 
-    pub fn enable_over8(&self) {
+    /// Makes call to control register one to enable oversampling by 8.
+    pub fn enable_over8(&mut self) {
         self.cr1.set_over8(true);
     }
 
-    pub fn disable_over8(&self) {
+    /// Makes call to control register one to disable oversampling by 8.
+    /// Will default to oversampling by 16.
+    pub fn disable_over8(&mut self) {
         self.cr1.set_over8(false);
     }
 
+    /// Makes call to control register one to check if over8 is enabled.
+    /// Returns true if enabled, false otherwise.
     pub fn get_over8(&self) -> bool {
         self.cr1.get_over8()
     }
 
-    pub fn set_hardware_flow_control(&self, hfc: HardwareFlowControl) {
+    // --------------------------------------------------------------
+    /// This section pertains to calls made to control register two (cr2).
```

Comment:
We definitely don't want this as a doc comment, it'll get documented on the next statement it sees.